packages/date-picker/test/validation.common.js

```diff
@@ -155,16 +155,10 @@ describe('validation', () => {
       datePicker.min = '2016-01-01';
       datePicker.max = '2016-12-31';
 
-      const invalidChangedSpy = sinon.spy();
-      datePicker.addEventListener('invalid-changed', invalidChangedSpy);
-
       await open(datePicker);
 
-      await waitForValueChange(datePicker, () => {
-        datePicker._overlayContent._selectDate(new Date('2017-01-01')); // Invalid
-      });
-
-      expect(invalidChangedSpy.calledOnce).to.be.true;
+      const selectResult = datePicker._overlayContent._selectDate(new Date('2017-01-01')); // Invalid
```

Comment:
Dropped the check for the invalid change since the new logic will prevent actually selecting an invalid date. Checking the boolean result of `_selectDate` should be sufficient. Same for below tests.